src/codegen/llvm.zig

```diff
@@ -553,7 +553,8 @@ const DataLayoutBuilder = struct {
             try self.typeAlignment(.float, 64, 64, 64, true, writer);
         }
         if (stack_abi != ptr_bit_width or self.target.cpu.arch == .msp430 or
```

Comment:
Not related to your PR, but these checks look really brittle. Is there any reason we can't just always trust `std.Target.stackAlignment()`? And if there is, shouldn't that be fixed? There's also a special-case for CSKY up on line 396. Weird situation overall.

Reply:
I got the same feeling. Given current code looks like one would need a lot of variant target for riscv feature combination, which would not feel right either. Guess would need a bit of code design change to allow this.